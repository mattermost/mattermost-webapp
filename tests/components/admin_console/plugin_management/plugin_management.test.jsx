// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';
import {shallow} from 'enzyme';

import PluginState from 'mattermost-redux/constants/plugins';

import PluginManagement from 'components/admin_console/plugin_management/plugin_management.jsx';

describe('components/PluginManagement', () => {
    const defaultProps = {
        config: {
            PluginSettings: {
                Enable: true,
                EnableUploads: true,
            },
        },
        pluginStatuses: {
            plugin_0: {
                id: 'plugin_0',
                version: '0.1.0',
                state: PluginState.PLUGIN_STATE_NOT_RUNNING,
                name: 'Plugin 0',
                description: 'The plugin 0.',
                is_prepackaged: false,
                active: false,
                instances: [
                    {
                        cluster_id: 'cluster_id_1',
                        state: PluginState.PLUGIN_STATE_NOT_RUNNING,
                        version: '0.1.0',
                    },
                ],
            },
            plugin_1: {
                id: 'plugin_1',
                version: '0.0.1',
                state: PluginState.PLUGIN_STATE_STOPPING,
                name: 'Plugin 1',
                description: 'The plugin.',
                is_prepackaged: false,
                active: true,
                instances: [
                    {
                        cluster_id: 'cluster_id_1',
                        state: PluginState.PLUGIN_STATE_NOT_RUNNING,
                        version: '0.0.1',
                    },
                    {
                        cluster_id: 'cluster_id_2',
                        state: PluginState.PLUGIN_STATE_RUNNING,
                        version: '0.0.2',
                    },
                ],
            },
        },
        plugins: {
            plugin_0: {
                active: false,
                description: 'The plugin 0.',
                id: 'plugin_0',
                name: 'Plugin 0',
                version: '0.1.0',
                settings_schema: {
                    footer: 'This is a footer',
                    header: 'This is a header',
                    settings: [],
                },
                webapp: {},
            },
            plugin_1: {
                active: true,
                description: 'The plugin 1.',
                id: 'plugin_1',
                name: 'Plugin 1',
                version: '0.1.0',
                settings_schema: {
                    footer: 'This is a footer',
                    header: 'This is a header',
                    settings: [],
                },
                webapp: {},
            },
        },
        actions: {
            uploadPlugin: jest.fn(),
            removePlugin: jest.fn(),
            getPluginStatuses: jest.fn().mockResolvedValue([]),
            enablePlugin: jest.fn(),
            disablePlugin: jest.fn(),
        },
    };

    test('should match snapshot', () => {
        const props = {...defaultProps};
        const wrapper = shallow(<PluginManagement {...props}/>);
        expect(wrapper).toMatchSnapshot();
    });

    test('should match snapshot, disabled', () => {
        const props = {
            ...defaultProps,
            config: {
                ...defaultProps.config,
                PluginSettings: {
                    ...defaultProps.config.PluginSettings,
                    Enable: false,
                },
            },
        };
        const wrapper = shallow(<PluginManagement {...props}/>);
        expect(wrapper).toMatchSnapshot();
    });

    test('should match snapshot, upload disabled', () => {
        const props = {
            ...defaultProps,
            config: {
                ...defaultProps.config,
                PluginSettings: {
                    ...defaultProps.config.PluginSettings,
                    EnableUploads: false,
                },
            },
        };
        const wrapper = shallow(<PluginManagement {...props}/>);
        expect(wrapper).toMatchSnapshot();
    });

    test('should match snapshot, No installed plugins', () => {
        const props = {
            config: {
                PluginSettings: {
                    Enable: true,
                    EnableUploads: true,
                },
            },
            pluginStatuses: {},
            plugins: {},
            actions: {
                uploadPlugin: jest.fn(),
                removePlugin: jest.fn(),
                getPluginStatuses: jest.fn().mockResolvedValue([]),
                enablePlugin: jest.fn(),
                disablePlugin: jest.fn(),
            },
        };
        const wrapper = shallow(<PluginManagement {...props}/>);
        wrapper.setState({loading: false});
        expect(wrapper).toMatchSnapshot();
    });

    test('should match snapshot, with installed plugins', () => {
        const wrapper = shallow(<PluginManagement {...defaultProps}/>);
        wrapper.setState({loading: false});
        expect(wrapper).toMatchSnapshot();
    });

    test('should match snapshot, with installed plugins and not settings link', () => {
        const props = {
            config: {
                PluginSettings: {
                    Enable: true,
                    EnableUploads: true,
                },
            },
            pluginStatuses: {
                plugin_0: {
                    id: 'plugin_0',
                    version: '0.1.0',
                    state: PluginState.PLUGIN_STATE_NOT_RUNNING,
                    name: 'Plugin 0',
                    description: 'The plugin 0.',
                    is_prepackaged: false,
                    active: false,
                    instances: [
                        {
                            cluster_id: 'cluster_id_1',
                            state: PluginState.PLUGIN_STATE_NOT_RUNNING,
                            version: '0.1.0',
                        },
                    ],
                },
                plugin_1: {
                    id: 'plugin_1',
                    version: '0.0.1',
                    state: PluginState.PLUGIN_STATE_STOPPING,
                    name: 'Plugin 1',
                    description: 'The plugin.',
                    is_prepackaged: false,
                    active: true,
                    instances: [
                        {
                            cluster_id: 'cluster_id_1',
                            state: PluginState.PLUGIN_STATE_NOT_RUNNING,
                            version: '0.0.1',
                        },
                        {
                            cluster_id: 'cluster_id_2',
                            state: PluginState.PLUGIN_STATE_RUNNING,
                            version: '0.0.2',
                        },
                    ],
                },
            },
            plugins: {
                plugin_0: {
                    active: false,
                    description: 'The plugin 0.',
                    id: 'plugin_0',
                    name: 'Plugin 0',
                    version: '0.1.0',
                    webapp: {},
                },
                plugin_1: {
                    active: true,
                    description: 'The plugin 1.',
                    id: 'plugin_1',
                    name: 'Plugin 1',
                    version: '0.1.0',
                    webapp: {},
                },
            },
            actions: {
                uploadPlugin: jest.fn(),
                removePlugin: jest.fn(),
                getPluginStatuses: jest.fn().mockResolvedValue([]),
                enablePlugin: jest.fn(),
                disablePlugin: jest.fn(),
            },
        };
        const wrapper = shallow(<PluginManagement {...props}/>);
        wrapper.setState({loading: false});
        expect(wrapper).toMatchSnapshot();
    });
});
