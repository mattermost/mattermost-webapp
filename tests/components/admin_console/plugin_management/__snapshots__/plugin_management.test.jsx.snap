// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/PluginManagement should match snapshot 1`] = `
<div
  className="wrapper--fixed"
>
  <h3
    className="admin-console-header"
  >
    <FormattedMessage
      defaultMessage="Management"
      id="admin.plugin.management.title"
      values={Object {}}
    />
  </h3>
  <form
    className="form-horizontal"
    role="form"
  >
    <div
      className="form-group"
    >
      <label
        className="control-label col-sm-4"
      >
        <FormattedMessage
          defaultMessage="Upload Plugin: "
          id="admin.plugin.uploadTitle"
          values={Object {}}
        />
      </label>
      <div
        className="col-sm-8"
      >
        <div
          className="file__upload"
        >
          <button
            className="btn btn-primary"
            disabled={false}
          >
            <FormattedMessage
              defaultMessage="Choose File"
              id="admin.plugin.choose"
              values={Object {}}
            />
          </button>
          <input
            accept=".gz"
            disabled={false}
            onChange={[Function]}
            type="file"
          />
        </div>
        <button
          className="btn"
          disabled={true}
          onClick={[Function]}
        >
          <FormattedMessage
            defaultMessage="Upload"
            id="admin.plugin.upload"
            values={Object {}}
          />
        </button>
        <div
          className="help-text no-margin"
        />
        <p
          className="help-text"
        >
          <InjectIntl(FormattedMarkdownMessage)
            defaultMessage="Upload a plugin for your Mattermost server. See [documentation](!https://about.mattermost.com/default-plugin-uploads) to learn more."
            id="admin.plugin.uploadDesc"
          />
        </p>
      </div>
    </div>
    <div
      className="form-group"
    >
      <label
        className="control-label col-sm-4"
      >
        <FormattedMessage
          defaultMessage="Installed Plugins: "
          id="admin.plugin.installedTitle"
          values={Object {}}
        />
      </label>
      <div
        className="col-sm-8"
      >
        <p
          className="help-text"
        >
          <FormattedHTMLMessage
            defaultMessage="Installed plugins on your Mattermost server. Pre-packaged plugins are installed by default, and can be disabled but not removed."
            id="admin.plugin.installedDesc"
            values={Object {}}
          />
        </p>
        <br />
      </div>
    </div>
  </form>
</div>
`;

exports[`components/PluginManagement should match snapshot, disabled 1`] = `
<div
  className="wrapper--fixed"
>
  <h3
    className="admin-console-header"
  >
    <FormattedMessage
      defaultMessage="Management"
      id="admin.plugin.management.title"
      values={Object {}}
    />
  </h3>
  <Banner
    description={
      <InjectIntl(FormattedMarkdownMessage)
        defaultMessage="Plugins are disabled on your server. To enable them, go to **Plugins > Configuration**."
        id="admin.plugin.management.banner"
      />
    }
    title={<div />}
  />
</div>
`;

exports[`components/PluginManagement should match snapshot, upload disabled 1`] = `
<div
  className="wrapper--fixed"
>
  <h3
    className="admin-console-header"
  >
    <FormattedMessage
      defaultMessage="Management"
      id="admin.plugin.management.title"
      values={Object {}}
    />
  </h3>
  <form
    className="form-horizontal"
    role="form"
  >
    <div
      className="form-group"
    >
      <label
        className="control-label col-sm-4"
      >
        <FormattedMessage
          defaultMessage="Upload Plugin: "
          id="admin.plugin.uploadTitle"
          values={Object {}}
        />
      </label>
      <div
        className="col-sm-8"
      >
        <div
          className="file__upload"
        >
          <button
            className="btn"
            disabled={true}
          >
            <FormattedMessage
              defaultMessage="Choose File"
              id="admin.plugin.choose"
              values={Object {}}
            />
          </button>
          <input
            accept=".gz"
            disabled={true}
            onChange={[Function]}
            type="file"
          />
        </div>
        <button
          className="btn"
          disabled={true}
          onClick={[Function]}
        >
          <FormattedMessage
            defaultMessage="Upload"
            id="admin.plugin.upload"
            values={Object {}}
          />
        </button>
        <div
          className="help-text no-margin"
        />
        <p
          className="help-text"
        >
          <InjectIntl(FormattedMarkdownMessage)
            defaultMessage="To enable plugin uploads, go to **Plugins > Configuration**. See [documentation](!https://about.mattermost.com/default-plugin-uploads) to learn more."
            id="admin.plugin.uploadDisabledDesc"
          />
        </p>
      </div>
    </div>
    <div
      className="form-group"
    >
      <label
        className="control-label col-sm-4"
      >
        <FormattedMessage
          defaultMessage="Installed Plugins: "
          id="admin.plugin.installedTitle"
          values={Object {}}
        />
      </label>
      <div
        className="col-sm-8"
      >
        <p
          className="help-text"
        >
          <FormattedHTMLMessage
            defaultMessage="Installed plugins on your Mattermost server. Pre-packaged plugins are installed by default, and can be disabled but not removed."
            id="admin.plugin.installedDesc"
            values={Object {}}
          />
        </p>
        <br />
      </div>
    </div>
  </form>
</div>
`;
